### PR TITLE
[kernel] implementation of checkCompatibleNSLaw  in OSNSP

### PR DESCRIPTION
**Pull request by vacary**

### CODE REVIEW

File: kernel/src/simulationTools/LinearOSNS.hpp [hunk 1]
```diff
@@ -266,6 +266,8 @@ class LinearOSNS : public OneStepNSProblem
     _keepLambdaAndYState = val ;
   }
 
+  virtual bool checkCompatibleNSLaw(NonSmoothLaw& nslaw) {return true;};
+
```
Comment (xhub) [inline kernel/src/simulationTools/LinearOSNS.hpp:270]: nitpick: why not a pure virtual so that a child class must implement this method?

**Comment by xhub** [comment]
The function `checkCompatibleNSLaw` has not been added to AVI, Relay, MLCP and others. Is it on worth doing it for all (and once and for all)?

File: kernel/src/simulationTools/LinearOSNS.hpp (hunk 1 above)
Comment (vacary) [reply kernel/src/simulationTools/LinearOSNS.hpp:270]: agree. Just I am lazy and I have not finished to implement all the classes. I will fix that at the end :-)

Comment (xhub) [reply kernel/src/simulationTools/LinearOSNS.hpp:270]: Sounds good :)

**Comment by vacary** [comment]
@xhub  Thanks for your review. I completed to PR in my last commit following your recommendation

**Comment by vacary** [comment]
Do not hesitate to check that I didn't forget some compatible nonsmooth laws.

**Comment by xhub** [comment]
I think that GeneralMechanical is missing, but this may be done in the other PR?

There is not "virtual" or "override" keyword added to `checkCompatibleNSLaw` in the child classes. My C++ is too rusty to remember if it does really matter. I don't think we expect to have child class of those. That my usual nitpicking :)

**Comment by vacary** [comment]
> I think that GeneralMechanical is missing, but this may be done in the other PR?

Yes, I added the method that do nothing

> There is not "virtual" or "override" keyword added to `checkCompatibleNSLaw` in the child classes. My C++ is too rusty to remember if it does really matter. I don't think we expect to have child class of those. That my usual nitpicking :)

For the moment, there is no second-level child classes, to I think it is sufficient. no ?

**Comment by xhub** [comment]
> > There is not "virtual" or "override" keyword added to `checkCompatibleNSLaw` in the child classes. My C++ is too rusty to remember if it does really matter. I don't think we expect to have child class of those. That my usual nitpicking :)
> 
> For the moment, there is no second-level child classes, to I think it is sufficient. no ?

Yes, I was nitpicking as usual :)

**Pull request merged**